### DISCUSSION TITLE
关于-度量类型和theme主题设置的bug

### DISCUSSION THREAD

**Discussion by cow-Rong** [antvis/G2#4970] [Q&A] [answered]
我设置了view的
theme: {
                columnWidthRatio: 1,
            }
但是这个柱图却显示有空隙
![image](https://user-images.githubusercontent.com/31033412/221596099-96891361-d1e1-4610-a0c4-856afa786f99.png)
debugger源码时候发现因为x轴是连续数字自动转成liner类型
所以我又设置x轴强制为离散的cat度量类型
 type: 'cat'
结果整个图形又错乱了
![image](https://user-images.githubusercontent.com/31033412/221598717-3098c3b7-8f13-4639-9493-f3264a36ce53.png)

请问各位大佬，有谁知道如何设置正确的度量类型吗？
##### Checklist

- [ ] `npm test` passes
- [ ] benchmarks are included
- [ ] commit message follows commit guidelines
- [ ] documents are updated

**Comment by hustcc** [comment, marked as answer]
在 4.x 中有几种解法：
1. 把 x 轴的数据转换成 string 类型的
2. 设置 x 轴 scale 类型为 cat

```ts
chart.scale('x-field', { type: 'cat' })
```
